### ISSUE TITLE
Unpacked TypedDict is not allowed as the first argument to Callable

### ISSUE THREAD

**Issue by Gobot1234** [microsoft/pyright#3599]
**Describe the bug**
Unpacked TypedDict is not allowed as the first argument to Callable

**To Reproduce**
```py
import typing_extensions
from collections.abc import Callable

class _FooKwargs(typing_extensions.TypedDict):
    bar: int

def foo(**kwargs: typing_extensions.Unpack[_FooKwargs]): ...  # fine

foo2: Callable[[typing_extensions.Unpack[_FooKwargs]], Any] = Any  # error
```

**Expected behavior**
No error
> Expected TypeVarTuple or Tuple as type argument for UnpackPylance[reportGeneralTypeIssues](https://github.com/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md#diagnostic-severity-rules)

**VS Code extension or command-line**
pylance v2022.6.20

**Comment by erictraut** [comment]
This is an expected error. This isn't a valid use of `Unpack`.

**Comment by erictraut** [comment]
`Unpack` works with a TypedDict only with `**kwargs`, and `Callable` doesn't support `**kwargs`. You could use it in a callback protocol, however.

```python
class MyCallback(Protocol):
    def __call__(self, **kwargs: Unpack[_FooKwargs]) -> Any: ...
```

**Comment by Gobot1234** [comment]
I would have liked to avoid this but yeah this does seem like the only way. Thanks for the fast response.
